zds/tutorialv2/management/commands/migrate_to_zep25.py

```diff
@@ -0,0 +1,269 @@
+        ],
+        # Autres
+        [
+            ('Arts, graphisme et multimédia', 'Graphisme 2D, Graphisme 3D, Musique, Son'),
+            ('Communication et management', 'Monde du travail'),
+            ('Zeste de Savoir', 'Zeste de Savoir'),
+            ('Autres', 'Littérature, Interview')
+        ]
+    ]
+    i = 0
+    for category in categories:
+        cat = Category()
+        cat.title = category
+        cat.description = category
+        cat.position = i
+        cat.slug = slugify(u'{}'.format(category))
+        cat.save()
+        print('[ZEP-25] : New category "{}" added'.format(cat))
+        for subcategory in subcategories[i]:
+            sub = SubCategory()
+            sub.title = subcategory[0]
+            sub.subtitle = subcategory[1]
+            sub.slug = slugify(u'{}'.format(subcategory[0]))
+            sub.save()
+            print('[ZEP-25] : New subcategory "{}" added'.format(sub))
+            catsubcat = CategorySubCategory()
+            catsubcat.category = cat
+            catsubcat.subcategory = sub
+            catsubcat.save()
+            print('[ZEP-25] : Relation "{}" <--> "{}" added'.format(cat, sub))
+        i += 1
+
+
+def alert_authors():
+    """
+    Send a private message to all authors (with a `PublishableContent`) for new tags system.
+
+    :return: None
+    """
+    bot = get_object_or_404(User, username=settings.ZDS_APP['member']['bot_account'])
+    users = []
+    contents = PublishableContent.objects.all()
+    for content in contents:
+        authors = content.authors.all()
+        for author in authors:
+            if author not in users:
+                users.append(author)
+    for user in users:
+        msg = 'Bonjour {0},\n\nDepuis la dernière version de Zeste de Savoir, tous les contenus (articles, tutoriels ' \
+              'et bientôt tribunes libres) possèdent maintenant une nouvelle classification ([ZEP-25](https://zestede' \
+              'savoir.com/forums/sujet/2378/zep-25-categorisation-des-articles-et-tutoriels/)). Les **tags** ont fait' \
+              ' leur apparition et les catégories ont été revues afin de faciliter et d\'aléger cette classification.' \
+              '\n\nLes anciennes catégories ont été transformées en tags et de nouvelles catégories plus générales on' \
+              't été ajoutés. L\'équipe de Zeste de Savoir va ou a déjà changé les catégories des contenus publiés.\n' \
+              '\nNous vous invitons à vérifier la catégorie de vos [articles](https://zestedesavoir.com/contenus/arti' \
+              'cles/{1}/) et [tutoriels](https://zestedesavoir.com/contenus/tutoriels/{1}/) mais également la pertine' \
+              'nce des tags et en ajouter si besoin.\n\n\nNous restons à votre disposition et votre écoute pour toute' \
+              's suggestions ou remarques,\n\nL\'équipe de Zeste de Savoir'
+        send_mp(
+            bot,
+            [user],
```

Comment:
oups dsl.

Le 24 mars 2016 à 10:06, Gérard Paligot <notifications@github.com> a écrit :

> In zds/tutorialv2/management/commands/migrate_to_zep25.py
> <https://github.com/zestedesavoir/zds-site/pull/3261#discussion_r57289286>
> :
>
> > +            if author not in users:
> > +                users.append(author)
> > +    for user in users:
> > +        msg = 'Bonjour {0},\n\nDepuis la dernière version de Zeste de Savoir, tous les contenus (articles, tutoriels ' \
> > +              'et bientôt tribunes libres) possèdent maintenant une nouvelle classification ([ZEP-25](https://zestede' \
> > +              'savoir.com/forums/sujet/2378/zep-25-categorisation-des-articles-et-tutoriels/)). Les **tags** ont fait' \
> > +              ' leur apparition et les catégories ont été revues afin de faciliter et d\'aléger cette classification.' \
> > +              '\n\nLes anciennes catégories ont été transformées en tags et de nouvelles catégories plus générales on' \
> > +              't été ajoutés. L\'équipe de Zeste de Savoir va ou a déjà changé les catégories des contenus publiés.\n' \
> > +              '\nNous vous invitons à vérifier la catégorie de vos [articles](https://zestedesavoir.com/contenus/arti' \
> > +              'cles/{1}/) et [tutoriels](https://zestedesavoir.com/contenus/tutoriels/{1}/) mais également la pertine' \
> > +              'nce des tags et en ajouter si besoin.\n\n\nNous restons à votre disposition et votre écoute pour toute' \
> > +              's suggestions ou remarques,\n\nL\'équipe de Zeste de Savoir'
> > +        send_mp(
> > +            bot,
> > +            [user],
>
> @artragis <https://github.com/artragis> On est dans le script de
> migration.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zestedesavoir/zds-site/pull/3261/files/753b2255672250bba3b649f7e2644c3579d54fdd#r57289286>
>